From 16323135a0923af1a1ce18d5f474cad3f4f6f941 Mon Sep 17 00:00:00 2001
From: chrfranke <chrfranke@4ea69e1a-61f1-4043-bf83-b5c94c648137>
Date: Sun, 4 Aug 2019 14:39:24 +0000
Subject: [PATCH] linux_nvme_ioctl.h: Replace with current version from Linux
 kernel sources (include/uapi/linux/nvme_ioctl.h fadccd8 2019-02-20). This
 version adds Linux-syscall-note to its GPL-2.0 (only) license. This should
 fix the GPL-2.0-or-later licensing problem (#1226).

git-svn-id: https://svn.code.sf.net/p/smartmontools/code@4935 4ea69e1a-61f1-4043-bf83-b5c94c648137
---
 trunk/smartmontools/ChangeLog          |  7 +++++++
 trunk/smartmontools/linux_nvme_ioctl.h | 11 ++---------
 2 files changed, 9 insertions(+), 9 deletions(-)

Index: ChangeLog
===================================================================
--- ChangeLog.orig
+++ ChangeLog
@@ -1,5 +1,12 @@
 $Id: ChangeLog 4883 2018-12-30 14:48:54Z chrfranke $
 
+2019-08-04  Christian Franke  <franke@computer.org>
+
+	linux_nvme_ioctl.h: Replace with current version from Linux kernel
+	sources (include/uapi/linux/nvme_ioctl.h fadccd8 2019-02-20).
+	This version adds Linux-syscall-note to its GPL-2.0 (only) license.
+	This should fix the GPL-2.0-or-later licensing problem (#1226).
+
 2018-12-30  Christian Franke  <franke@computer.org>
 
 	smartmontools 7.0
Index: linux_nvme_ioctl.h
===================================================================
--- linux_nvme_ioctl.h.orig
+++ linux_nvme_ioctl.h
@@ -1,15 +1,7 @@
+/* SPDX-License-Identifier: GPL-2.0 WITH Linux-syscall-note */
 /*
  * Definitions for the NVM Express ioctl interface
  * Copyright (c) 2011-2014, Intel Corporation.
- *
- * This program is free software; you can redistribute it and/or modify it
- * under the terms and conditions of the GNU General Public License,
- * version 2, as published by the Free Software Foundation.
- *
- * This program is distributed in the hope it will be useful, but WITHOUT
- * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
- * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
- * more details.
  */
 
 #ifndef _UAPI_LINUX_NVME_IOCTL_H
@@ -61,5 +53,6 @@ struct nvme_passthru_cmd {
 #define NVME_IOCTL_IO_CMD	_IOWR('N', 0x43, struct nvme_passthru_cmd)
 #define NVME_IOCTL_RESET	_IO('N', 0x44)
 #define NVME_IOCTL_SUBSYS_RESET	_IO('N', 0x45)
+#define NVME_IOCTL_RESCAN	_IO('N', 0x46)
 
 #endif /* _UAPI_LINUX_NVME_IOCTL_H */
